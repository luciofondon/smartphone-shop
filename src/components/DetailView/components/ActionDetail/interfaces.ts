import { SelectChangeEvent } from '@mui/material';
import { IProductAdd } from '../../../Root/interfaces';
import { IProductDetail } from '../../interfaces';

export interface IActionDetail {
  product: IProductDetail;
  handleClickAddProduct: (product: IProductAdd) => void;
}

export interface IUseActionDetail {
  product: IProductDetail;
  handleClickAddProduct: (product: IProductAdd) => void;
}

export interface IUseActionDetailResponse {
  states: {
    storage: number;
    color: number;
  };
  actions: {
    handleSubmit: (event: React.FormEvent<HTMLFormElement>) => void;
    handleChange: (event: SelectChangeEvent) => void;
  };
}
