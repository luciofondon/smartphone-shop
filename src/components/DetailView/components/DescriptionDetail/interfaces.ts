import { IProductDetail } from '../../interfaces';

export interface IDescriptionDetail {
  product: IProductDetail;
}
