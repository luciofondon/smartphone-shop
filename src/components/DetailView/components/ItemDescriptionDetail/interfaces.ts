export interface IItemDescriptrionDetail {
  label: string;
  value: string;
}
