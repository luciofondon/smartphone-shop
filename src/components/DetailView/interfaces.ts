import { IProductAdd } from '../Root/interfaces';

interface IColor {
  code: number;
  name: string;
}

interface IStorage {
  code: number;
  name: string;
}

export interface IProductDetail {
  announced: string;
  audioJack: string;
  battery: string;
  bluetooth: string[];
  brand: string;
  chipset: string;
  colors: string[];
  cpu: string;
  dimentions: string;
  displayResolution: string;
  displaySize: string;
  displayType: string;
  edge: string;
  externalMemory: string;
  gprs: string;
  gps: string;
  gpu: string;
  id: string;
  imgUrl: string;
  internalMemory: string[];
  model: string;
  networkSpeed: string;
  networkTechnology: string;
  nfc: string;
  options: { colors: IColor[]; storages: IStorage[] };
  os: string;
  price: string;
  primaryCamera: string[];
  radio: string;
  ram: string;
  secondaryCmera: string[];
  sensors: string[];
  sim: string;
  speaker: string;
  status: string;
  usb: string;
  weight: string;
  wlan: string[];
}

export interface IUseDetailViewResponse {
  states: {
    product: IProductDetail | null;
  };
  actions: {
    handleClickAddProduct: (product: IProductAdd) => void;
  };
}
