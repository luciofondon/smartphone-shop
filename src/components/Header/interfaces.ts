export interface IHeader {
  totalProducts: number;
}
