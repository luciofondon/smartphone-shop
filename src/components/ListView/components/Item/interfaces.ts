export interface IProduct {
  id: string;
  model: string;
  brand: string;
  imgUrl: string;
  price: number;
}

export interface IItem {
  product: IProduct;
}
