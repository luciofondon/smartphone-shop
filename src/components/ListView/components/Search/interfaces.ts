export interface ISearch {
  handleChangeFilter: (text: string) => void;
}
