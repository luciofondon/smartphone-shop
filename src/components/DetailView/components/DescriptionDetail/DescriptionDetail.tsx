import React from 'react';
import ItemDescriptionDetail from '../ItemDescriptionDetail';

import { IDescriptionDetail } from './interfaces';
import './styles.css';

function DescriptionDetail({ product }: IDescriptionDetail) {
  return (
    <div>
      <ItemDescriptionDetail label="Marca" value={product.brand} />
      <ItemDescriptionDetail label="Modelo" value={product.model} />
      <ItemDescriptionDetail label="Precio" value={product.price} />
      <ItemDescriptionDetail label="CPU" value={product.brand} />
      <ItemDescriptionDetail label="RAM" value={product.cpu} />
      <ItemDescriptionDetail label="Sistema operativo" value={product.os} />
      <ItemDescriptionDetail label="Resolucion" value={product.displayResolution} />
      <ItemDescriptionDetail label="Batería" value={product.battery} />
      {product.primaryCamera && product.primaryCamera.length > 0 && (
        <ItemDescriptionDetail label="Cámara principal" value={product.primaryCamera[0]} />
      )}
      {product.primaryCamera && product.primaryCamera.length > 0 && (
        <ItemDescriptionDetail label="Cámara secundaria" value={product.secondaryCmera[0]} />
      )}
      <ItemDescriptionDetail label="Cámara secundaria" value={product.secondaryCmera[0]} />
      <ItemDescriptionDetail label="Dimensiones" value={product.dimentions} />
      <ItemDescriptionDetail label="Peso" value={product.weight} />
    </div>
  );
}

export default DescriptionDetail;
