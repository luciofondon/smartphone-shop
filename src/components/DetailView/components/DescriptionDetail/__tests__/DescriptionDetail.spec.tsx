import React from 'react';
import { render } from '@testing-library/react';
import DescriptionDetail from '../DescriptionDetail';
import { IProductDetail } from '../../../interfaces';

test('render DescriptionDetail', () => {
  const productMock: IProductDetail = {
    announced: '',
    audioJack: '',
    battery: 'Removable Li-Po 4080 mAh battery',
    bluetooth: [],
    brand: 'Alcatel',
    chipset: '',
    colors: [],
    cpu: 'Octa-core 1.3 GHz Cortex-A53',
    dimentions: '153.8 x 75.6 x 8.5 mm (6.06 x 2.98 x 0.33 in)',
    displayResolution: '5.5 inches ',
    displaySize: '',
    displayType: '',
    edge: '',
    externalMemory: '',
    gprs: '',
    gps: '',
    gpu: '',
    id: '',
    imgUrl: 'https://front-test-api.herokuapp.com/images/P2oqviM96_ozwsgZkj9Xf.jpg',
    internalMemory: [],
    model: 'Liquid Z6 Plus',
    networkSpeed: '',
    networkTechnology: '',
    nfc: '',
    options: {
      colors: [{ code: 1, name: 'Rojo' }],
      storages: [{ code: 1, name: '100GB' }],
    },
    os: '',
    price: '220',
    primaryCamera: ['13 MPautofocusLED flash'],
    radio: '',
    ram: '3 GB RAM',
    secondaryCmera: ['5 MP'],
    sensors: [],
    sim: '',
    speaker: '',
    status: '',
    usb: '',
    weight: '169',
    wlan: [],
  };
  const { container } = render(<DescriptionDetail product={productMock} />);
  expect(container).toMatchSnapshot();
});
