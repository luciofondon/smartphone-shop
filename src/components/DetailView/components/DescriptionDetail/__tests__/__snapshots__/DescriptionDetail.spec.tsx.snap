// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render DescriptionDetail 1`] = `
<div>
  <div>
    <div
      class="item-description"
    >
      <div
        class="title-description"
      >
        Marca
      </div>
      <div
        class="value-description"
      >
        Alcatel
      </div>
    </div>
    <div
      class="item-description"
    >
      <div
        class="title-description"
      >
        Modelo
      </div>
      <div
        class="value-description"
      >
        Liquid Z6 Plus
      </div>
    </div>
    <div
      class="item-description"
    >
      <div
        class="title-description"
      >
        Precio
      </div>
      <div
        class="value-description"
      >
        220
      </div>
    </div>
    <div
      class="item-description"
    >
      <div
        class="title-description"
      >
        CPU
      </div>
      <div
        class="value-description"
      >
        Alcatel
      </div>
    </div>
    <div
      class="item-description"
    >
      <div
        class="title-description"
      >
        RAM
      </div>
      <div
        class="value-description"
      >
        Octa-core 1.3 GHz Cortex-A53
      </div>
    </div>
    <div
      class="item-description"
    >
      <div
        class="title-description"
      >
        Sistema operativo
      </div>
      <div
        class="value-description"
      />
    </div>
    <div
      class="item-description"
    >
      <div
        class="title-description"
      >
        Resolucion
      </div>
      <div
        class="value-description"
      >
        5.5 inches 
      </div>
    </div>
    <div
      class="item-description"
    >
      <div
        class="title-description"
      >
        Batería
      </div>
      <div
        class="value-description"
      >
        Removable Li-Po 4080 mAh battery
      </div>
    </div>
    <div
      class="item-description"
    >
      <div
        class="title-description"
      >
        Cámara principal
      </div>
      <div
        class="value-description"
      >
        13 MPautofocusLED flash
      </div>
    </div>
    <div
      class="item-description"
    >
      <div
        class="title-description"
      >
        Cámara secundaria
      </div>
      <div
        class="value-description"
      >
        5 MP
      </div>
    </div>
    <div
      class="item-description"
    >
      <div
        class="title-description"
      >
        Cámara secundaria
      </div>
      <div
        class="value-description"
      >
        5 MP
      </div>
    </div>
    <div
      class="item-description"
    >
      <div
        class="title-description"
      >
        Dimensiones
      </div>
      <div
        class="value-description"
      >
        153.8 x 75.6 x 8.5 mm (6.06 x 2.98 x 0.33 in)
      </div>
    </div>
    <div
      class="item-description"
    >
      <div
        class="title-description"
      >
        Peso
      </div>
      <div
        class="value-description"
      >
        169
      </div>
    </div>
  </div>
</div>
`;
